firmware/cmake/stmlib.cmake

```diff
@@ -55,16 +55,19 @@ message("  🔃 Registered stm32f412rx_cube_library() function")
 # HAL_SRCS: the ones that we want, stripped prefixes
 # SYSCALLS: most of the functions defined inside are weak references, only used to make sure it builds without error.
 # USB_ENABLED: flags if usb middleware should be included.
+# USB_ENABLED: flags if usb middleware should be included.
 function(stm32f412rx_cube_library
         HAL_LIB_NAME
         HAL_CONF_DIR
         HAL_SRCS
         SYSCALLS
         IOC_CHECKSUM
         USB_ENABLED
+        USB_ENABLED
 )
     set(DRIVERS_DIR "${STM32CUBEF4_SOURCE_DIR}/Drivers")
     set(FREERTOS_DIR "${STM32CUBEF4_SOURCE_DIR}/Middlewares/Third_Party/FreeRTOS/Source")
+    set(USB_MIDDLEWARE_DIR "${STM32CUBEF4_SOURCE_DIR}/Middlewares/ST/STM32_USB_Device_Library")
```

Comment:
There's a lot of changes here due to merge conflicts -  I recommend copying the origin/master `firmware/cmake/stmlib.cmake` into here - should drop these diffs.